drivers/i2c/i2c_nrfx_twi.c

```diff
@@ -122,9 +123,26 @@ static const struct i2c_driver_api i2c_nrfx_twi_driver_api = {
 	.transfer  = i2c_nrfx_twi_transfer,
 };
 
-static int init_twi(struct device *dev, const nrfx_twi_config_t *config)
+static int init_twi(struct device *dev, const nrfx_twi_config_t *config,
+		    u32_t bitrate)
 {
-	nrfx_err_t result = nrfx_twi_init(&get_dev_config(dev)->twi, config,
+	nrfx_twi_config_t cfg = *config;
+
+	switch (bitrate) {
+	case I2C_BITRATE_STANDARD:
+		cfg.frequency = NRF_TWI_FREQ_100K;
```

Comment:
Could we use a little trick instead and do this translation at compile time? I mean something like:
```
#define I2C_NRFX_TWI_INVALID_FREQUENCY  ((nrf_twi_frequency_t)-1)
#define I2C_NRFX_TWI_FREQUENCY(bitrate)					    \
	( bitrate == I2C_BITRATE_STANDARD ? NRF_TWI_FREQ_100K		    \
	: bitrate == 250000               ? NRF_TWI_FREQ_250K		    \
	: bitrate == I2C_BITRATE_FAST     ? NRF_TWI_FREQ_400K		    \
					  : I2C_NRFX_TWI_INVALID_FREQUENCY)
```
and in `I2C_NRFX_TWI_DEVICE(idx)`:
```
		const nrfx_twi_config_t config =			       \
		{							       \
			.scl       = DT_NORDIC_NRF_I2C_I2C_##idx##_SCL_PIN,    \
			.sda       = DT_NORDIC_NRF_I2C_I2C_##idx##_SDA_PIN,    \
			.frequency = I2C_NRFX_TWI_FREQUENCY(		       \
				DT_NORDIC_NRF_I2C_I2C_##idx##_CLOCK_FREQUENCY) \
		};							       \
		if (config.frequency == I2C_NRFX_TWI_INVALID_FREQUENCY) {      \
			return -EINVAL;					       \
		}							       \
		return init_twi(dev, &config);				       \
```
Even a moderately smart compiler should be able to optimize out this checking of the frequency when it is valid (which we normally expect to be).
This way `config` will stay `const` as it was, and no extra runtime code will be added to handle its data.
What you think?

Reply:
Looks great, I switched to this method